lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -248,13 +264,14 @@ def update(self, subsection, block_structure=None):
         if not PersistentGradesEnabledFlag.feature_enabled(self.course.id):
             return
 
-        block_structure = self._get_block_structure(block_structure)
-        subsection_grade = SubsectionGrade(subsection, self.course)
-        subsection_grade.compute(self.student, block_structure, self._scores_client, self._submissions_scores)
+        with persistence_safe_fallback():
+            block_structure = self._get_block_structure(block_structure)
```

Comment:
This line looks like it could be outside the contextmanager.  

Also _get_block_structure() is a weird method.  It's kind of a validator, and kind of a default-value-setter, but it's definitely not a getter.